resources/analysisTools/indelCallingWorkflow/checkSampleSwap_TiN.pl

```diff
@@ -294,21 +315,42 @@
 #######################################
 ### Finding and plotting TiN
 
-print "Rscript $TiN_R -f $snvsGT_germlineRare_txt --oPlot $snvsGT_germlineRare_png --oFile $snvsGT_germlineRare_oFile -p $pid --chrLength $chrLengthFile --cFunction $canopy_Function --SeqType $seqType --rightBorder $rightBorder --bottomBorder $bottomBorder --vcf $snvsGT_germlineRare --Ovcf $snvsGT_germlineRare_oVCF\n";
+my $runRscript_code  = "Rscript $TiN_R -f $snvsGT_germlineRare_txt";
+   $runRscript_code .= " --oPlot $snvsGT_germlineRare_png";
+   $runRscript_code .= " --oFile $snvsGT_germlineRare_oFile";
+   $runRscript_code .= " -p $pid";
+   $runRscript_code .= " --chrLength $chrLengthFile";
+   $runRscript_code .= " --cFunction $canopy_Function";
+   $runRscript_code .= " --SeqType $seqType";
+   $runRscript_code .= " --rightBorder $rightBorder";
+   $runRscript_code .= " --bottomBorder $bottomBorder";
+   $runRscript_code .= " --vcf $snvsGT_germlineRare";
+   $runRscript_code .= " --Ovcf $snvsGT_germlineRare_oVCF";
 
-my $runRscript = system("Rscript $TiN_R -f $snvsGT_germlineRare_txt --oPlot $snvsGT_germlineRare_png --oFile $snvsGT_germlineRare_oFile -p $pid --chrLength $chrLengthFile --cFunction $canopy_Function --SeqType $seqType --rightBorder $rightBorder --bottomBorder $bottomBorder --vcf $snvsGT_germlineRare --Ovcf $snvsGT_germlineRare_oVCF\n");
+print $runRscript_code, "\n";
+
+my $runRscript = system($runRscript_code);
 
 if($runRscript != 0) { 
   `rm $jsonFile`;
-  die "Error while running $TiN_R in swapChecker\n";
+  if($runRscript == 50) {
+    $! = 50;
+    die "Less than 50 rare germline variants, might be the control and tumor didn't match - Sample swapped. Exiting the analysis. $!";
+  }
+  else{
+    $! = $runRscript;
+    die "Error while running $TiN_R in swapChecker. $!";
+  }
 }
  
 chomp($json{'tindaGermlineRareAfterRescue'} = `cat $snvsGT_germlineRare_oFile | grep 'Germline' | wc -l`);
 chomp($json{'tindaSomaticAfterRescue'}  = `cat $snvsGT_germlineRare_oFile | grep 'Somatic_Rescue' | wc -l`);
 
 if($json{'tindaSomaticAfterRescue'} > 0) {
 
-  $json{'tindaSomaticAfterRescueMedianAlleleFreqInControl'} = `cat $snvsGT_germlineRare_oFile | grep 'Somatic_Rescue' | cut -f5 | sort -n | awk ' { a[i++]=\$1;} END { x=int((i+1)/2); if (x < (i+1)/2) print (a[x-1]+a[x])/2; else print a[x-1]; }'`;
+  $json{'tindaSomaticAfterRescueMedianAlleleFreqInControl'} = `cat $snvsGT_germlineRare_oFile | \\ 
```

Comment:
Odd, to fill a value in a Perl script by calling an awk command. Should this not get substituted by Perl?